packages/order-experience-api/src/api/createController.ts

```diff
@@ -43,30 +49,39 @@ export class CreateController extends AbstractController<typeof requestSchema> {
     req: ValidatedRequest<typeof requestSchema>,
     res: Response
   ): Promise<any> {
-    const { namespace, user } = req.body
+    const { namespace, user, paymentFilters } = req.body
     logger.debug(`Create Order for namespace ${namespace} and user ${user}`)
-    const dto = new Data(await createOrder({ namespace, user }))
+    const orderData = await createOrder({ namespace, user })
+    const paymentFiltersData = await savePaymentFiltersAdmin(paymentFilters)
+    const dto = new Data({
+      ...orderData,
+      paymentFilters: paymentFiltersData,
+    })
     return this.created<any>(res, dto.serialize())
   }
 
   protected async createWithItems(
     req: ValidatedRequest<typeof requestSchema>,
     res: Response
   ): Promise<any> {
-    const { namespace, user, items, customer } = req.body
+    const { namespace, user, items, customer, paymentFilters } = req.body
     logger.debug(
       `Create Order with Items for namespace ${namespace} and user ${user}`
     )
-    const dto = new Data(
-      await createOrderWithItems({
-        namespace,
-        user,
-        items,
-        customer,
-        ...calculateTotalsFromItems({ items }),
-      })
-    )
 
+    const orderData = await createOrderWithItems({
+      namespace,
+      user,
+      items,
+      customer,
+      ...calculateTotalsFromItems({ items }),
+    })
+    const paymentFilterData = await savePaymentFiltersAdmin(paymentFilters)
```

Comment:
What happens if paymentFilters array is empty/ null? 

Does this endpoint still try to call savePaymentFiltersAdmin?

Reply:
Yes it will call the endpoint. I can add an if -statement to enable saving only when paymentFilters are present.